Added IP based access control support

mohitsinghnegi2 opened the pull request:
# Description

Currently, the ML Default Workspace and AI Studio Hub Workspace offer two network access options: All Network and Disable This PR introduces a third option i.e. Enabled from selected IP addresses, using new field called network ACLs (Network Access Control Lists) in Workspace Class. This field will be use to define IP Address/ IP ranges on workspace. That would mean Only defined IP addresses/ranges would get access to the Default / Hub workspaces.

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

mohitsinghnegi2 committed:
refactor code

mohitsinghnegi2 committed:
refactor code

mohitsinghnegi2 committed:
Adding IP based access control to SDK

mohitsinghnegi2 committed:
Added test and change logs

mohitsinghnegi2 committed:
Examples to choose one of three Public network access settings

mohitsinghnegi2 committed:
resolved circular dependency

mohitsinghnegi2 committed:
resolved circular dependency

mohitsinghnegi2 committed:
Added Ip based access control support to hub workspace

mohitsinghnegi2 committed:
Merge branch 'main' of https://github.com/Azure/azure-sdk-for-python into azure-sdk-support-for-networkAcls
Pulled latest changes from master

mohitsinghnegi2 committed:
updated changelog file

mohitsinghnegi2 committed:
removed ipallowlist dependencey

mohitsinghnegi2 committed:
Merge branch 'main' of https://github.com/Azure/azure-sdk-for-python into azure-sdk-support-for-networkAcls
pull main branch changes

mohitsinghnegi2 committed:
resolved ManagedServiceIdentity version icompatibility issue

mohitsinghnegi2 committed:
fixed breaking test

mohitsinghnegi2 committed:
reformatted code

mohitsinghnegi2 committed:
reformatted code

doubleyewdee commented on sdk/ml/azure-ai-ml/azure/ai/ml/_schema/workspace/network_acls.py:
```diff
@@ -0,0 +1,63 @@
+        :type data: OrderedDict[str, Any]
+        :raises ValidationError: If the schema is invalid.
+        """
+        if data["default_action"] not in [NetworkAcls.DefaultActionType.DENY, NetworkAcls.DefaultActionType.ALLOW]:
```
```suggestion
        if data["default_action"] not in set(NetworkAcls.DefaultActionType.DENY, NetworkAcls.DefaultActionType.ALLOW):
```

doubleyewdee commented on sdk/ml/azure-ai-ml/azure/ai/ml/entities/__init__.py:
```diff
@@ -158,41 +121,26 @@
 from ._job.input_port import InputPort
 from ._job.job import Job
 from ._job.job_limits import CommandJobLimits
 from ._job.job_resource_configuration import JobResourceConfiguration
```
Lots of no-op import changes that look like either isort settings changed, or you're not using some isort configuration as intended? You might want to check on this.

mohitsinghnegi2 committed:
Merge branch 'main' of https://github.com/Azure/azure-sdk-for-python into azure-sdk-support-for-networkAcls

mohitsinghnegi2 committed:
removed doc example

mohitsinghnegi2 committed:
Merge branch 'main' of https://github.com/Azure/azure-sdk-for-python into azure-sdk-support-for-networkAcls
merge branch

mohitsinghnegi2 committed:
code refactor

mohitsinghnegi2 committed:
Fixed Generate API Stubs issue

mohitsinghnegi2 committed:
Fixed Generate API Stubs issue

mohitsinghnegi2 committed:
Merge branch 'main' of https://github.com/Azure/azure-sdk-for-python into azure-sdk-support-for-networkAcls

mohitsinghnegi2 committed:
refactor code

mohitsinghnegi2 committed:
add doc string

mohitsinghnegi2 committed:
add doc string

mohitsinghnegi2 committed:
add doc string

mohitsinghnegi2 committed:
add doc string

mohitsinghnegi2 committed:
add doc string

mohitsinghnegi2 committed:
add doc string

azure-sdk committed:
[AutoRelease] t2-connectedcache-2024-11-19-72393(can only be merged by SDK owner) (#38605)

* code and test

* update testcases

* update CHANGELOG.md

---------

Co-authored-by: azure-sdk <PythonSdkPipelines>
Co-authored-by: ChenxiJiang333 <v-chenjiang@microsoft.com>

azure-sdk committed:
[AutoRelease] t2-terraform-2024-11-05-91454(can only be merged by SDK owner) (#38326)

* code and test

* update testcases

* Update CHANGELOG.md

* Update CHANGELOG.md

---------

Co-authored-by: azure-sdk <PythonSdkPipelines>
Co-authored-by: ChenxiJiang333 <v-chenjiang@microsoft.com>
Co-authored-by: ChenxiJiang333 <119990644+ChenxiJiang333@users.noreply.github.com>

msyyc committed:
Delete scripts/typespec_refresh_sdk directory (#38682)

msyyc committed:
Delete sdk/containerregistry/azure-mgmt-containerregistry/azure/mgmt/containerregistry/v2018_02_01_preview directory (#38688)

After https://github.com/Azure/azure-rest-api-specs/pull/28811 merged, 2018-02 was retired and we need to delete the directory manually otherwise the Python SDK generation will be broken.

mohitsinghnegi2 committed:
Merge branch 'main' of https://github.com/Azure/azure-sdk-for-python into azure-sdk-support-for-networkAcls

mohitsinghnegi2 committed:
removed managed identity related changes

mohitsinghnegi2 committed:
remove formatting changes to _credentials.py file

mohitsinghnegi2 committed:
Update _credentials.py

mohitsinghnegi2 committed:
Update _credentials.py
